examples/balanced_lineupsNBA.py

```diff
@@ -0,0 +1,118 @@
+"""
+Generate as many lineups as you want with
+max exposures of your selected exposure to any one player,
+with a specified minimum average of your choice.
+The output here is a CSV file that can be used
+to upload lineups at https://www.draftkings.com/lineup/upload
+
+Pre-reqs:
+1) Salary file exists at passed location
+2) Projection file exists at passed location
+3) PIDS file downloaded from aforementioned location
+
+Example usage:
+```python
+from examples import balanced_lineupsNBA
+balanced_lineupsNBA.run()
+```
+"""
+
+import random
+import numpy
+from collections import Counter
+from optimize import run as optimizer_run
+from argparse import Namespace
+
+from csv_upload.nba_upload import (
+    update_upload_csv,
+    create_upload_file,
+    map_pids,
+)
+
+DEFAULT_ARGS = dict(
+    dtype='wr',
```

Comment:
`dtype`, `duo`, `season`, `no_double_te` and `historical` are not relevant for NBA - I would change to `None` and move them to the bottom of this dict.

Reply:
Looks like it's breaking when `duo` is set to `None`. Line 212 in `optimize.py`. I'll need to dig into it a little.

Reply:
Darn, this shouldn't happen; no need to address in this PR - I can address separately.